src/main/java/com/microsoft/graph/tasks/LargeFileUploadTask.java

```diff
@@ -0,0 +1,323 @@
+                    UploadResult<T> result;
+                    result = uploadSliceAsync(request, exceptionsList);
+                    amountUploaded += request.getRangeLength();
+                    if(progress != null) {
+                        progress.report(amountUploaded, this.totalUploadLength);
+                    }
+                    if (result.isUploadSuccessful()) {
+                        return CompletableFuture.completedFuture(result);
+                    }
+                }
+                updateSessionStatusAsync().get();
+                uploadTries += 1;
+                if (uploadTries < maxTries) {
+                    TimeUnit.SECONDS.sleep((long) 2 * uploadTries * uploadTries);
+                }
+            } catch (InterruptedException ex) {
+                Thread.currentThread().interrupt();
+                CompletableFuture<UploadResult<T>> exceptionalResult = new CompletableFuture<>();
+                exceptionalResult.completeExceptionally(ex);
+                return exceptionalResult;
+            } catch (IOException| ExecutionException| ServiceException ex) {
+                CompletableFuture<UploadResult<T>> exceptionalResult = new CompletableFuture<>();
+                exceptionalResult.completeExceptionally(ex);
+                return exceptionalResult;
+            }
+
+        }
+        CompletableFuture<UploadResult<T>> exceptionalResult = new CompletableFuture<>();
+        exceptionalResult.completeExceptionally(new CancellationException());
+        return exceptionalResult;
+    }
+    /**
+     * Resume the upload task.
+     * @return An UploadResult model containing the information from the server resulting from the upload request.
+     */
+    @Nonnull
+    public CompletableFuture<UploadResult<T>> resumeAsync() {
+        return this.resumeAsync(3, null);
+    }
+    /**
+     * Resume the upload task.
+     * @param maxTries Number of times to retry the task before giving up.
+     * @param progress IProgress interface describing how to report progress.
+     * @return An UploadResult model containing the information from the server resulting from the upload request..
+     */
+    @Nonnull
+    public CompletableFuture<UploadResult<T>> resumeAsync(int maxTries, @Nullable IProgressCallback progress) {
+        IUploadSession session;
+        try{
+            session = updateSessionStatusAsync().get();
+        } catch (ExecutionException ex) {
+            CompletableFuture<UploadResult<T>> exceptionalResult = new CompletableFuture<>();
+            exceptionalResult.completeExceptionally(ex);
+            return exceptionalResult;
+        } catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
+            CompletableFuture<UploadResult<T>> exceptionalResult = new CompletableFuture<>();
+            exceptionalResult.completeExceptionally(ex);
+            return exceptionalResult;
```

Comment:
wouldn't it be better to simply let that exception flow all the way to the caller?

Reply:
@baywet Are you suggesting that I remove the try-catch block and simply let the task throw an InterruptedException? Sonar cloud was flagging for a critical issue without Thread.currentThread().Interrupt() inside of a InterruptedException try-catch block. 
https://stackoverflow.com/questions/4906799/why-invoke-thread-currentthread-interrupt-in-a-catch-interruptexception-block

Reply:
I'm suggesting to remove only that catch block, not the rest of the try catch

Reply:
please do. Thanks for checking